src/app/applications/components/utils.tsx

```diff
@@ -6,7 +6,20 @@ import { AppContext } from '../../shared/context';
 import * as appModels from '../../shared/models';
 import { services } from '../../shared/services';
 
-export async function deleteApplication(appName: string, context: AppContext, success: () => void) {
+export async function syncApplication(appName: string, revision: string, prune: boolean, context: AppContext) {
+    try {
+        await services.applications.sync(appName, revision, prune);
+    } catch (e) {
+        context.apis.notifications.show({
+            content: <ErrorNotification title='Unable to deploy revision' e={e}/>,
+            type: NotificationType.Error,
+        });
+    } finally {
```

Comment:
Refresh should be done after operation is completed, not after it is requested. It should be done by app controller. Please remove force refresh